schema/schema.graphql

```diff
@@ -10,6 +10,8 @@ const chats = require("./chats/chat.graphql");
 const utils = require("./utils.graphql");
 
 const typeDefs = gql`
+  directive @auth on FIELD_DEFINITION 
```

Comment:
Can we classify the directive so that schema can be updated accordingly?

Example:

```
directive @auth (
  requires: Role = Admin,
) on FIELD_DEFINITION 

enum Role {
  ADMIN
  USER
  UNKNOWN
}
```

After this, we can have schema like this in future:
```
  type Organization @auth(requires: USER) {
    image:String
    _id: ID!
    name:String!
    description: String!
    isPublic: Boolean! 
    creator: User!
    members: [User]
    admins(adminId: ID): [User] @auth(requires: ADMIN)
    membershipRequests: [MembershipRequest] @auth(requires: ADMIN)
    blockedUsers: [User] @auth(requires: ADMIN)
    visibleInSearch: Boolean!
    apiUrl:String!
  }
```

Reply:
This will restrict individual parameter of a node accordingly

Reply:
Isn't this more applicable to #298? In the current resolver, I am only checking for `authenticated` routes. Or should I consolidate both the functionalities in a single directive?

Reply:
I will make this change on a new pr :+1: